### PR TITLE
[SPARK-44544][INFRA][FOLLOWUP] Force run `run_python_packaging_tests`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
run `run_python_packaging_tests` when there are any changes in PySpark

### Why are the changes needed?
https://github.com/apache/spark/pull/42146 make CI run `run_python_packaging_tests` only within `pyspark-errors` (see https://github.com/apache/spark/actions/runs/5666118302/job/15359190468 and https://github.com/apache/spark/actions/runs/5668071930/job/15358091003)

image

but I ignored that `pyspark-errors` maybe skipped (because no related source changes), so the `run_python_packaging_tests` maybe also skipped  unexpectedly (see https://github.com/apache/spark/actions/runs/5666523657/job/15353485731)

image

this PR is to run `run_python_packaging_tests` even if `pyspark-errors` is skipped

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
updated CI

### CODE REVIEW

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon @LuciferYang @panbingkun

File: dev/run-tests.py [hunk 1]
```diff
@@ -559,6 +559,10 @@ def main():
 
         changed_modules = test_modules
         if len(changed_modules) == 0:
+            # If SKIP_PACKAGING is explicitly set (no matter true or false),
+            # let run_python_packaging_tests to determine whether it should be run.
+            if not os.environ.get("SKIP_PYTHON") and os.environ.get("SKIP_PACKAGING") is not None:
+                run_python_packaging_tests()
```
Comment (HyukjinKwon) [inline dev/run-tests.py:565]: Hm, to run pip packaging test, Spark has to be built first but here is before we build. I think it won't work correctly?

Comment (zhengruifeng) [reply dev/run-tests.py:565]: oops, let me double check, thanks!

File: dev/sparktestsupport/modules.py [hunk 2]
```diff
@@ -1040,7 +1040,14 @@ def __hash__(self):
 pyspark_errors = Module(
     name="pyspark-errors",
     dependencies=[],
-    source_file_regexes=["python/pyspark/errors"],
+    source_file_regexes=[
+        # [SPARK-44544] Force the execution of pyspark_errors when there is any changes
+        # in PySpark, since the Python Packaging Tests is only enabled within this module.
+        # This module is the smallest Python test modules, it only contains 1 test file
+        # and normally take < 2 seconds, so the additional cost is small.
+        "python/",
```
Comment (zhengruifeng) [inline dev/sparktestsupport/modules.py:1048]: cc @HyukjinKwon I find it is pretty hard to me to decouple the packaging tests from other pyspark tests. what about this workaround?

Comment (HyukjinKwon) [reply dev/sparktestsupport/modules.py:1048]: SGTM

File: dev/sparktestsupport/modules.py [hunk 3]
```diff
@@ -1040,7 +1040,14 @@ def __hash__(self):
 pyspark_errors = Module(
     name="pyspark-errors",
     dependencies=[],
-    source_file_regexes=["python/pyspark/errors"],
+    source_file_regexes=[
+        # [SPARK-44544] Force the execution of pyspark_errors when there is any changes
```
Comment (HyukjinKwon) [inline dev/sparktestsupport/modules.py:1044]:
```suggestion
        # SPARK-44544: Force the execution of pyspark_errors when there is any changes
```

**Review by HyukjinKwon — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Review by panbingkun — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thank you guys!

merged to master/branch-3.5/branch-3.4

**Pull request closed without merging**